app/controllers/cohorts/columns_controller.rb

```diff
@@ -28,9 +28,35 @@ def update
         end
       end
       @cohort.update(column_state: columns)
+      if params.include? :translation
+        translate_columns(params[:translation])
+      end
       respond_with(@cohort, location: cohort_path(@cohort))
     end
 
+    def translate_columns(translations)
+      return unless can_edit_translations?
+
+      columns = cohort_source.available_columns
+      locale = 'en'
+      columns.each do |column|
+        if column.attributes.include?(:translation_key)
+          proposed_translation = translations[column.column]
+          key = column.translation_key
+          translation_key = TranslationKey.find_or_create_by(key: key)
+          existing_translation = translation_key.translations.find_by(locale: locale)
+          if existing_translation.present? && existing_translation != proposed_translation
```

Comment:
I believe first_or_create will simplify this quite a bit.  You can pass it a block for the update.